compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -41,6 +44,38 @@ impl MetaVarExpr {
         check_trailing_token(&mut tts, psess)?;
         let mut iter = args.trees();
         let rslt = match ident.as_str() {
+            "concat" => {
+                fn element<'psess>(
+                    iter: &mut RefTokenTreeCursor<'_>,
+                    psess: &'psess ParseSess,
+                    span: Span,
+                ) -> PResult<'psess, MetaVarExprConcatElem> {
+                    let is_var = try_eat_dollar(iter);
+                    let ident = parse_ident(iter, psess, span)?;
+                    Ok(if is_var {
+                        MetaVarExprConcatElem::Var(ident)
+                    } else {
+                        MetaVarExprConcatElem::Ident(ident)
+                    })
+                }
+
+                let mut result = Vec::new();
+                loop {
+                    result.push(element(&mut iter, psess, ident.span)?);
```

Comment:
The fallback span should point to the closing parenthesis here.
The same span can be used in a couple of places below too.

Reply:
Well, I am now using a span that ranges until the closing brace if that makes sense